libsplinter/src/registry/client/mod.rs

```diff
@@ -0,0 +1,110 @@
+// Copyright 2018-2021 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Traits and implementations useful for communicating with the registry as a client.
+
+use std::fmt;
+
+mod reqwest;
+
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+
+use crate::error::InternalError;
+
+pub use self::reqwest::ReqwestRegistryClient;
+
+pub trait RegistryClient {
+    /// Adds a node to the registry.
+    ///
+    /// # Arguments
+    ///
+    /// * `node` - The node to be added
+    fn add_node(&self, node: &RegistryNode) -> Result<(), InternalError>;
+
+    /// Retrieves a node from the registry.
+    ///
+    /// # Arguments
+    ///
+    /// * `identity` - The id of the node to be retrieved
+    fn get_node(&self, identity: &str) -> Result<Option<RegistryNode>, InternalError>;
+
+    /// Lists the nodes in the registry.
+    ///
+    /// # Arguments
+    ///
+    /// * `filter` - Filter to apply to the list of nodes
+    fn list_nodes(&self, filter: Option<&str>) -> Result<RegistryNodeListSlice, InternalError>;
+
+    /// Update a node in the registry.
+    ///
+    /// # Arguments
+    ///
+    /// * `node` - The updated node replacing the current node in the registry
+    fn update_node(&self, node: &RegistryNode) -> Result<(), InternalError>;
+
+    /// Delete a node from the registry.
+    ///
+    /// # Arguments
+    ///
+    /// * `identity` - The id of the node to be deleted
+    fn delete_node(&self, identity: &str) -> Result<(), InternalError>;
+}
+
+#[cfg(feature = "registry")]
```

Comment:
This cfg and the one below are unnecessary as the feature guard exists around the entire module.

Reply:
That got copied over from the original client, removed